packages/console/src/pages/EnterpriseSsoDetails/Settings/index.tsx

```diff
@@ -0,0 +1,266 @@
+  onUpdated: (data: SsoConnectorWithProviderConfig) => void;
+};
+
+export type FormType = Pick<SsoConnector, 'branding' | 'connectorName'> & {
+  syncProfile: SyncProfileMode;
+  domains: MultiInputOption[];
+};
+
+const duplicatedDomainsErrorCode = 'single_sign_on.duplicated_domains';
+const forbiddenDomainsErrorCode = 'single_sign_on.forbidden_domains';
+
+const dataToFormParser = (data: SsoConnectorWithProviderConfig) => {
+  const { branding, connectorName, domains, syncProfile } = data;
+  return {
+    branding,
+    connectorName,
+    domains: domains.map((domain) => ({ value: domain, id: generateStandardShortId() })),
+    syncProfile: syncProfile ? SyncProfileMode.EachSignIn : SyncProfileMode.OnlyAtRegister,
+  };
+};
+
+const formDataToSsoConnectorParser = (
+  formData: FormType
+): Pick<SsoConnector, 'branding' | 'connectorName' | 'domains' | 'syncProfile'> => {
+  const { branding, connectorName, domains, syncProfile } = formData;
+  return {
+    branding,
+    connectorName,
+    domains: domains.map(({ value }) => value),
+    syncProfile: syncProfile === SyncProfileMode.EachSignIn,
+  };
+};
+
+function Settings({ data, isDeleted, onUpdated }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { isReady: isUserAssetsServiceReady } = useUserAssetsService();
+  const api = useApi({ hideErrorToast: true });
+
+  const syncProfileOptions = [
+    {
+      value: SyncProfileMode.OnlyAtRegister,
+      title: t('enterprise_sso_details.sync_profile_option.register_only'),
+    },
+    {
+      value: SyncProfileMode.EachSignIn,
+      title: t('enterprise_sso_details.sync_profile_option.each_sign_in'),
+    },
+  ];
+
+  const formMethods = useForm<FormType>({ defaultValues: dataToFormParser(data) });
+  const {
+    control,
+    watch,
+    setValue,
+    setError,
+    clearErrors,
+    handleSubmit,
+    register,
+    formState: { isDirty, isSubmitting, errors },
+    reset,
+  } = formMethods;
+
+  const onSubmit = handleSubmit(
+    trySubmitSafe(async (formData: FormType) => {
+      if (isSubmitting) {
+        return;
+      }
+
+      try {
+        const updatedSsoConnector = await api
+          .patch(`api/sso-connectors/${data.id}`, { json: formDataToSsoConnectorParser(formData) })
+          .json<SsoConnectorWithProviderConfig>();
+
+        reset(dataToFormParser(updatedSsoConnector));
+        toast.success(t('general.saved'));
+        onUpdated(updatedSsoConnector);
+      } catch (error: unknown) {
+        /**
+         * Should render invalid domains:
+         * - show `error` tag for forbidden domains
+         * - show `info` tag for duplicated domains
+         *
+         * Also manually passed the returned error message to `setError` to show the error message in-place.
+         */
+        if (error instanceof HTTPError) {
+          const { response } = error;
+          const metadata = await response.json<RequestErrorBody<{ data: string[] }>>();
```

Comment:
```suggestion
          const metadata = await response.clone().json<RequestErrorBody<{ data: string[] }>>();
```
Use `clone()` or we will encounter a reding content error when we try to read the response content in later logic.